client/promotions/spotlight/index.tsx

```diff
@@ -97,7 +96,13 @@ const SpotlightPromotion: React.FC = () => {
 	};
 
 	const handleDismiss = () => {
-		dismissPromotion( promotionId as string );
+		if ( ! spotlightVariation ) {
```

Comment:
```suggestion
		if ( ! promotionId || ! spotlightVariation ) {
```
What do you think about adding a check to ensure promotionId isn’t null or undefined as well?

Reply:
Good idea! Updated in 6bfd88ad4